### PR TITLE
[SPARK-12324][MLLIB][DOC] Fixes the sidebar in the ML documentation

### PR DESCRIPTION
**Pull request by thunterdb**
This fixes the sidebar, using a pure CSS mechanism to hide it when the browser's viewport is too narrow.
Credit goes to the original author @Titan-C (mentioned in the NOTICE).

Note that I am not a CSS expert, so I can only address comments up to some extent.

Default view:
screen shot 2015-12-14 at 12 46 39 pm

When collapsed manually by the user:
screen shot 2015-12-14 at 12 54 02 pm

Disappears when column is too narrow:
screen shot 2015-12-14 at 12 47 22 pm

Can still be opened by the user if necessary:
screen shot 2015-12-14 at 12 51 15 pm

### CODE REVIEW

**Comment by thunterdb** [comment]
@jkbradley can you take a look at this fix?

**Comment by jkbradley** [comment]
Tested locally.  Functionally, this seems fine.  I'll rely on someone else to check the CSS.

File: docs/_layouts/global.html [hunk 1]
```diff
@@ -128,19 +128,31 @@
 
             {% if page.url contains "/ml" %}
               {% include nav-left-wrapper-ml.html nav-mllib=site.data.menu-mllib nav-ml=site.data.menu-ml %}
+              <input id="nav-trigger" class="nav-trigger" checked="" type="checkbox">
+              <label for="nav-trigger"></label>
+                <div class="content-with-sidebar" id="content">
```
Comment (parano) [inline docs/_layouts/global.html]: indent

File: docs/css/main.css [hunk 2]
```diff
@@ -171,24 +202,104 @@ a.anchorjs-link:hover { text-decoration: none; }
...
+.nav-trigger + label:after {
+    content: '»';
+}
+
+label {
+  z-index: 10;
+}
+
+label[for="nav-trigger"] {
+    position: fixed;
```
Comment (parano) [inline docs/css/main.css]: I saw both 2 and 4 space indent in this file, it should be consistant

File: docs/_layouts/global.html [hunk 3]
```diff
@@ -128,19 +128,31 @@
 
             {% if page.url contains "/ml" %}
               {% include nav-left-wrapper-ml.html nav-mllib=site.data.menu-mllib nav-ml=site.data.menu-ml %}
+              <input id="nav-trigger" class="nav-trigger" checked="" type="checkbox">
```
Comment (parano) [inline docs/_layouts/global.html]: ```checked=""``` could be just ```checked```

File: docs/css/main.css [hunk 4]
```diff
@@ -39,13 +39,26 @@
...
+  color: #1D1F22;
+  max-width: 1024px;
+  margin-top: 10px;
+  margin-left: auto;
+  margin-right: auto;
+  border-radius: 15px;
+  position: relative;
+}
+
+body #contentREMOVE {
```
Comment (parano) [inline docs/css/main.css]: is that a typo? ```#contentREMOVE```

File: docs/css/main.css [hunk 5]
```diff
@@ -171,24 +202,104 @@ a.anchorjs-link:hover { text-decoration: none; }
...
+.nav-trigger {
+    position: fixed;
+    clip: rect(0, 0, 0, 0);
+}
+
+.nav-trigger + label:after {
+    content: '»';
+}
+
+label {
```
Comment (parano) [inline docs/css/main.css]: why not put this in the next selector ```label[for="nav-trigger"]```

File: docs/css/main.css [hunk 6]
```diff
@@ -171,24 +202,104 @@ a.anchorjs-link:hover { text-decoration: none; }
...
+    position: fixed;
+    margin-left: 0px;
+    padding-top: 100px;
+    padding-left: 5px;
+    width: 10px;
+    height: 80%;
+    cursor: pointer;
+    background-size: contain;
+    background-color: #D4F0FF;
+    border-radius: 10px 0px 0px 10px;
```
Comment (parano) [inline docs/css/main.css]: the border-radius looks a bit wired since it's the only thing that has rounded corners in the page, try remove this line?

**Comment by parano** [comment]
@thunterdb @jkbradley I left some comments on the css part, otherwise looks good to me

File: docs/_layouts/global.html (hunk 1 above)
Comment (thunterdb) [inline docs/_layouts/global.html]: Fixed

File: docs/css/main.css (hunk 2 above)
Comment (thunterdb) [inline docs/css/main.css]: Fixed everywhere to 2 spaces

File: docs/css/main.css (hunk 4 above)
Comment (thunterdb) [inline docs/css/main.css]: Thanks!

File: docs/css/main.css (hunk 5 above)
Comment (thunterdb) [inline docs/css/main.css]: Good question. I tried to do this and it failed to take the z-index into account despite my best efforts. This is the only way I found to make it work. If you think it is too wide, then I can try to add a CSS class to it.

File: docs/css/main.css (hunk 6 above)
Comment (thunterdb) [inline docs/css/main.css]: Removing the border.

File: docs/css/main.css [hunk 7]
```diff
@@ -101,6 +98,26 @@ a:hover code {
   max-width: 914px;
 }
 
+.content {
+  z-index: 1;
+  position: relative;
+  background-color: #FFF;
```
Comment (jkbradley) [inline docs/css/main.css]: 2 instances of background-color here

File: docs/css/main.css [hunk 8]
```diff
@@ -101,6 +98,26 @@ a:hover code {
...
+  max-width: 914px;
+  line-height: 1.6; /* Inspired by Github's wiki style */
+  background-color: white;
+  padding-left: 15px;
+}
+
+.content-with-sidebar {
+  z-index: 1;
+  position: relative;
+  background-color: #FFF;
```
Comment (jkbradley) [inline docs/css/main.css]: ditto

**Comment by jkbradley** [comment]
Functionally, it seems good to me.  (tested locally)

File: docs/css/main.css (hunk 7 above)
Comment (thunterdb) [inline docs/css/main.css]: oops thanks!

File: docs/css/main.css (hunk 8 above)
Comment (thunterdb) [inline docs/css/main.css]: Done

**Comment by jkbradley** [comment]
LGTM.  I'll merge this with master and branch-1.6

**Pull request closed without merging**